login-module/src/core.js

```diff
@@ -3,84 +3,74 @@ import Quiz from './quiz'
+	const opts = { ...options }
 	const hasRequiredOptions = requiredOptions.every(item => opts[item])
 	if (!hasRequiredOptions) {
 		throw new Error(`required option missing - one of ${requiredOptions.join(', ')}`)
 	}
 	return opts
 }
 
-class GraphAcademyServices {
+export default class GraphAcademyCore {
 	constructor(options = {}) {
 		const requiredOptions = ['trainingClassName', 'enrollmentUrl', 'authResult']
-		this.options = validateOptions(options, requiredOptions, {})
-		const { stage, trainingClassName, authResult } = this.options
-		this.authResult = authResult
+		const defaultOptions = {
+			stage: 'prod',
+		}
+		this.options = { ...defaultOptions, ...validateOptions(options, requiredOptions) }
+		this.webAuth = new WebAuth({
+			clientID: 'hoNo6B00ckfAoFVzPTqzgBIJHFHDnHYu',
+			domain: 'login.neo4j.com',
+			redirectUri: `${window.location.origin}/accounts/login`,
+			audience: 'neo4j://accountinfo/',
+			scope: 'read:account-info openid email profile user_metadata',
+			responseType: 'token id_token'
+		})
+		const { stage, trainingClassName } = this.options
 		this.certificate = new Certificate(trainingClassName, stage)
 		this.enrollment = new Enrollment(trainingClassName, stage)
 		this.quiz = new Quiz(trainingClassName, stage)
+
 	}
 
-	async enrollStudentInClass (firstName, lastName) {
-		return this.enrollment.enrollStudentInClass(firstName, lastName, this.getAccessToken())
+	async login() {
+		const { webAuth } = this
+		webAuth.checkSession({}, (err, result) => {
+			if (result) this.authResult = result;
+			return [err, result];
+		})
+	}
+
+	async enrollStudentInClass(firstName, lastName) {
+		this.validateRequest();
+		return this.enrollment.enrollStudentInClass(firstName, lastName, this.authResult.accessToken)
 	}
 
 	async getEnrollmentForClass() {
-		return this.enrollment.getEnrollmentForClass(this.getAccessToken())
+		this.validateRequest();
+		return this.enrollment.getEnrollmentForClass(this.authResult.accessToken)
 	}
 
 	async getQuizStatus() {
-		return this.quiz.getQuizStatus(this.getAccessToken())
+		this.validateRequest();
+		return this.quiz.getQuizStatus(this.authResult.accessToken)
 	}
 
 	async getClassCertificate() {
-		return this.certificate.getClassCertificate(this.getAccessToken())
+		this.validateRequest();
+		return this.certificate.getClassCertificate(this.authResult.accessToken)
 	}
 
 	async postQuizStatus(passed, failed) {
-		return this.quiz.postQuizStatus(passed, failed, this.getAccessToken())
-	}
-
-	getAccessToken () {
-		return this.authResult.accessToken
+		this.validateRequest();
+		return this.quiz.postQuizStatus(passed, failed, this.authResult.accessToken)
 	}
-}
 
-export default class GraphAcademyCore {
-	constructor(options = {}) {
-		const requiredOptions = ['trainingClassName', 'enrollmentUrl']
-		const defaultOptions = {
-			stage: 'prod',
-			trainingClassName: null,
-			enrollmentUrl: null,
+	validateRequest() {
+		if (!this.authResult.accessToken) {
```

Comment:
I guess it's a reasonable trade-off to avoid a wrapping class around the `accessToken`. I think we should need to check if `this.authResult` is defined:
```suggestion
		if (!this.authResult || !this.authResult.accessToken) {
```